pkg/kubectl/apis/manifest/v1alpha1/types.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Package contains all the metadata of the package and drives package searching and browsing,
+// and support the fork/rebase upgrade workflow. It can be used by something like an app registry.
+type Package struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// The name of the package.
+	// The name of an individual instance should live in ObjectMeta.Name.
```

Comment:
FYI, we're going to want to generate user-facing documentation from these comments, so referring to json names rather than Go types and fields is preferred.